arangod/Aql/AttributeNamePath.h

```diff
@@ -0,0 +1,123 @@
+////////////////////////////////////////////////////////////////////////////////
+/// DISCLAIMER
+///
+/// Copyright 2014-2016 ArangoDB GmbH, Cologne, Germany
+/// Copyright 2004-2014 triAGENS GmbH, Cologne, Germany
+///
+/// Licensed under the Apache License, Version 2.0 (the "License");
+/// you may not use this file except in compliance with the License.
+/// You may obtain a copy of the License at
+///
+///     http://www.apache.org/licenses/LICENSE-2.0
+///
+/// Unless required by applicable law or agreed to in writing, software
+/// distributed under the License is distributed on an "AS IS" BASIS,
+/// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+/// See the License for the specific language governing permissions and
+/// limitations under the License.
+///
+/// Copyright holder is ArangoDB GmbH, Cologne, Germany
+///
+/// @author Jan Steemann
+////////////////////////////////////////////////////////////////////////////////
+
+#ifndef ARANGOD_AQL_ATTRIBUTE_NAME_PATH_H
+#define ARANGOD_AQL_ATTRIBUTE_NAME_PATH_H 1
+
+#include <functional>
+#include <string>
+#include <vector>
+
+namespace arangodb {
+namespace aql {
+
+/// @brief helper class to handle top level and nested attribute paths.
+/// a top-level attribute (e.g. _key) will be stored in a vector with a single
+/// element. nested attributes (e.g. a.b.c) will be stored in a vector with
+/// multiple elements
+struct AttributeNamePath {
```

Comment:
didn't we have something like this for index fields ?

Reply:
Yes, we have `AttributeName` in lib/Basics/AttributeNameParser.h, but that is slightly different. It supports attribute name expansion (e.g. `a.b[*].c`), which means one cannot simply created a vector of strings but only of {string, bool} pairs. This makes it really annoying and more complicated to use in cases when just non-expanded attribute names are needed.

I can fuse the two classes, making things a bit more inefficient and harder to use, for the benefit of having less code around. Not sure what is better here.